### PR TITLE
log writing

### PR DESCRIPTION
**Pull request by tomyyyD**
updating to write logs after creating the directory to store logs in.

### CODE REVIEW

File: state_machine/applications/flight/Tasks/beacon_task.py [hunk 1]
```diff
@@ -24,5 +24,10 @@ async def main_task(self):
         except Exception:
             try:
                 os.mkdir('/sd/logs/')
+                # make sure to still log file when making the directory or else we lose the first beacon_packet of data
+                beacon_packet = logs.beacon_packet(self)
+                file = open(f"/sd/logs/log{int(currTime//TIMEINTERVAL)}.txt", "ab+")
```
Comment (jbwillis) [inline state_machine/applications/flight/Tasks/beacon_task.py:29]: I think this filename should include the word "beacon" somewhere

**Review by jbwillis — CHANGES_REQUESTED** [review]

File: state_machine/applications/flight/Tasks/beacon_task.py (hunk 1 above)
Comment (jbwillis) [reply state_machine/applications/flight/Tasks/beacon_task.py:29]: same with the one above

**Comment by tomyyyD** [comment]
line 30 of the beacon_task file is a line that I had to add for testing on the emulation. When on the hardware it will already have the "/sd" folder that is why it is commented out at the moment, but remember to uncomment if you are testing on emulation.

File: state_machine/applications/flight/Tasks/beacon_task.py (hunk 1 above)
Comment (tomyyyD) [reply state_machine/applications/flight/Tasks/beacon_task.py]: Sounds good! I have changed the files to be beacon_log instead of just log.

**Review by jbwillis — CHANGES_REQUESTED** [review]

File: state_machine/applications/flight/Tasks/beacon_task.py [hunk 2]
```diff
@@ -15,14 +15,25 @@ async def main_task(self):
...
-            file = open(f"/sd/logs/log{int(currTime//TIMEINTERVAL)}.txt", "ab+")
+            file = open(current_file, "ab+")
             file.write(bytearray(beacon_packet))
             file.close()
         except Exception:
             try:
-                os.mkdir('/sd/logs/')
+                # This line to be removed when testing on hardware which already has the sd directory
+                # leave this file in when testing on emulation
+                # os.mkdir('/sd')
```
[the remark this answers is not in the document]
Comment (jbwillis) [reply state_machine/applications/flight/Tasks/beacon_task.py:30]: Something like 

Uncomment the following for logging with emulator

**Review by jbwillis — APPROVED** [review]

**Review by jbwillis — CHANGES_REQUESTED** [review]
Actually could you make the file naming match the way it is done in the log task?

**Review by thetazero — APPROVED** [review]
Made the changes requested

**Comment by thetazero** [comment]
Tested on hardware and added a mkdir -p equivalent to make it more robust.

**Pull request merged**